packages/it-multipart/index.js

```diff
@@ -85,31 +110,56 @@ async function * yieldUntilAfter (haystack, needle) {
   }
 }
 
+/**
+ * @param {PrefixStream<Buffer>} haystack
+ * @param {Buffer} needle
+ * @returns {Promise<void>}
+ */
 async function consumeUntilAfter (haystack, needle) {
-  for await (const chunk of yieldUntilAfter(haystack, needle)) { // eslint-disable-line no-unused-vars
+  for await (const _chunk of yieldUntilAfter(haystack, needle)) { // eslint-disable-line no-unused-vars
 
   }
 }
 
-// a stream that lets us push content back into it for consumption elsewhere
+/**
+ * @template T
+ * @typedef {Object} PrefixPush
+ * @property {function(T):void} push
+ */
+
+/**
+ * @template T
+ * @typedef {It<T> & PrefixPush<T>} PrefixStream
+ */
+
+/**
+ * a stream that lets us push content back into it for consumption elsewhere
+ * @param {AsyncIterable<Buffer>} stream
+ * @returns {PrefixStream<Buffer>}
+ */
 function prefixStream (stream) {
+  /** @type {Buffer[]} */
   const buffer = []
   const streamIterator = stream[Symbol.asyncIterator]()
 
   const iterator = {
     [Symbol.asyncIterator]: () => {
       return iterator
     },
-    next: () => {
+    next: async () => {
```

Comment:
This function has no `await`, why add the `async`?

Reply:
Because async iterator supposed to return `Promise<{done:true}|{done:false, value:T}>`. Most things would work if you don't wrap in promise, but valid use e.g. `iterator.next().then(...)` would throw.